src/controllers/controllerAuth.class.php

```diff
@@ -112,4 +130,165 @@ public function logOut() : void {
+     */
+    public function register(?string $username, ?string $email, ?string $password): void {
+        if(session_status() === PHP_SESSION_NONE) { session_start(); }
+        // Vérifier si l'email, le nom d'utilisateur et le mot de passe sont valides
+        if ($this->validateUsername($username) &&
+            $this->validateEmail($email) &&
+            $this->validatePassword($password)) {
+
+            // Initialiser la variable de message de résultat
+            $resultMessage = null;
+
+            // Hash le mot de passe
+            $hashedPassword = password_hash($password, PASSWORD_DEFAULT);
+
+
+            $userDAO = new UserDAO($this->getPdo());
+            $playerDAO = new PlayerDAO($this->getPdo());
+            
+            // Vérifier si l'utilisateur et le joueur existent
+            $existingUser = $userDAO->findByEmail($email) !== null;
+            $existingPlayer = $playerDAO->findByUsername($username) !== null;
+            
+            $resultUser = false;
+            
+            // Si l'utilisateur et le joueur n'existent pas, créer l'utilisateur
+            if (!$existingUser && !$existingPlayer)
+            {
+                $emailVerifToken = bin2hex(random_bytes(30)); // Générer un token de vérification de l'email
+                $resultUser = $userDAO->createUser($email, $hashedPassword, $emailVerifToken);
+
+                // Envoi du mail avec phpmailer
+                $mail = new PHPMailer(true); // Création d'un objet PHPMailer
+                try {
+                    // Configuration technique
+                    $mail->isSMTP(); // Utilisation du protocole SMTP
+                    $mail->Host = MAIL_HOST; // Hôte du serveur SMTP
+                    $mail->SMTPAuth = true; // Authentification SMTP
+                    $mail->SMTPSecure = MAIL_SECURITY; // Cryptage SMTP
+                    $mail->Port = MAIL_PORT; // Port SMTP
+
+                    // Configuration de l'authentification
+                    $mail->Username = MAIL_USER; // Nom d'utilisateur de l'expéditeur
+                    $mail->Password = MAIL_PASS; // Mot de passe de l'expéditeur
+                    $mail->setFrom(MAIL_FROM); // Adresse de l'expéditeur
+                    $mail->addAddress($email); // Adresse du destinataire
+
+                    // Configuration du message
+                    $mail->isHTML(true); // Utilisation du format HTML pour le corps du message
+                    $mail->Subject = 'Confirmation de votre compte Comus Party' . MAIL_BASE; // Sujet du message
+                    $mail->Body = // Corps du message
+                        '<p>Vous avez créer un compte sur le site Comus.</p>
+                        <p>Pour confirmer votre compte, cliquez sur le lien ci-dessous.</p>
+                        <a href="http://localhost:8000/confirm-email/' . urlencode($emailVerifToken) . '"><button>Confirmer mon compte</button></a>';
```

Comment:
Ne pas oublier que cette url ne marche que sur ton pc. 
Il faut utiliser la variable d'environnement `BASE_URL`

```suggestion
                        <a href="' . $BASE_URL . '/confirm-email/' . urlencode($emailVerifToken) . '"><button>Confirmer mon compte</button></a>';
```